src/openforms/submissions/api/permissions.py

```diff
@@ -105,3 +105,37 @@ class DownloadSubmissionReportPermission(TimestampedTokenPermission):
 
 class SubmissionStatusPermission(TimestampedTokenPermission):
     token_generator = submission_status_token_generator
+
+
+class CanNavigateBetweenSubmissionStepsPermission(permissions.BasePermission):
+    """
+    Check if the user is allowed to interact with a submission step if the previous submission steps are not completed.
+    """
+
+    def has_object_permission(
+        self, request: Request, view: APIView, obj: SubmissionStep
+    ) -> bool:
+        user = request.user
+
+        order = obj.form_step.order
+        if user.is_superuser or (
+            user.is_staff
+            and user.has_perm("accounts.can_navigate_between_submission_steps")
+        ):
+            return True
```

Comment:
Do we need a new permission for this? Can we not tie to `user.is_staff` or the existing `forms.can_change_form` permissions?